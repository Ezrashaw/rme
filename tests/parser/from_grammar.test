fn w()=true*(((--false*f(t,4,t))));-true*(v)/u+(Tci!-z)*true*lxN;-(FI/t(LO,-b==O-4-U==(true==.7==CE)))/2.;