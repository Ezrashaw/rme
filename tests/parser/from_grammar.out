[36;1mfn[0m w[95m([0m[95m)[0m[90;1m = [0m[36;1mtrue[0m * [95m([0m[95m([0m[95m([0m--[36;1mfalse[0m * f[95m([0mt[90;1m, [0m[33;3m4[0m[90;1m, [0mt[95m)[0m[95m)[0m[95m)[0m[95m)[0m[90;1m;[0m
-[36;1mtrue[0m * [95m([0mv[95m)[0m / u + [95m([0mTci! - z[95m)[0m * [36;1mtrue[0m * lxN[90;1m;[0m
-[95m([0mFI / t[95m([0mLO[90;1m, [0m-b == O - [33;3m4[0m - U == [95m([0m[36;1mtrue[0m == [33;3m0.7[0m == CE[95m)[0m[95m)[0m[95m)[0m / [33;3m2[0m[90;1m;[0m

---AST Statement (semi: 34..35) #0---
Statement(FnDef)@0..34
| VarDef@0..34
| | fn_kw: 0..2
| | name: "w"@3..4
| | parens: (4..5, 5..6)
| | equals: 6..7
| | expr: BinaryOp@7..34
| | | op: Mul@11..12
| | | lhs: Bool(true)@7..11
| | | rhs: Paren@12..34
| | | | parens: (12..13, 33..34)
| | | | expr: Paren@13..33
| | | | | parens: (13..14, 32..33)
| | | | | expr: Paren@14..32
| | | | | | parens: (14..15, 31..32)
| | | | | | expr: BinaryOp@15..31
| | | | | | | op: Mul@22..23
| | | | | | | lhs: UnaryOp@15..22
| | | | | | | | op: Negation@15..16
| | | | | | | | expr: UnaryOp@16..22
| | | | | | | | | op: Negation@16..17
| | | | | | | | | expr: Bool(false)@17..22
| | | | | | | rhs: FunctionCall@23..31
| | | | | | | | call-expr: Variable("f")@23..24
| | | | | | | | parens: (24..25, 30..31)
| | | | | | | | arg #0 (comma): Some(26..27)
| | | | | | | | arg #0: Variable("t")@25..26
| | | | | | | | arg #1 (comma): Some(28..29)
| | | | | | | | arg #1: Float(4.0)@27..28
| | | | | | | | arg #2 (comma): None
| | | | | | | | arg #2: Variable("t")@29..30
-------

---AST Statement (semi: 64..65) #1---
Statement(Expression)@35..64
| BinaryOp@35..64
| | op: Add@46..47
| | lhs: BinaryOp@35..46
| | | op: Div@44..45
| | | lhs: BinaryOp@35..44
| | | | op: Mul@40..41
| | | | lhs: UnaryOp@35..40
| | | | | op: Negation@35..36
| | | | | expr: Bool(true)@36..40
| | | | rhs: Paren@41..44
| | | | | parens: (41..42, 43..44)
| | | | | expr: Variable("v")@42..43
| | | rhs: Variable("u")@45..46
| | rhs: BinaryOp@47..64
| | | op: Mul@60..61
| | | lhs: BinaryOp@47..60
| | | | op: Mul@55..56
| | | | lhs: Paren@47..55
| | | | | parens: (47..48, 54..55)
| | | | | expr: BinaryOp@48..54
| | | | | | op: Sub@52..53
| | | | | | lhs: UnaryOp@48..52
| | | | | | | op: Factorial@51..52
| | | | | | | expr: Variable("Tci")@48..51
| | | | | | rhs: Variable("z")@53..54
| | | | rhs: Bool(true)@56..60
| | | rhs: Variable("lxN")@61..64
-------

---AST Statement (semi: 105..106) #2---
Statement(Expression)@65..105
| BinaryOp@65..105
| | op: Div@102..103
| | lhs: UnaryOp@65..102
| | | op: Negation@65..66
| | | expr: Paren@66..102
| | | | parens: (66..67, 101..102)
| | | | expr: BinaryOp@67..101
| | | | | op: Div@69..70
| | | | | lhs: Variable("FI")@67..69
| | | | | rhs: FunctionCall@70..101
| | | | | | call-expr: Variable("t")@70..71
| | | | | | parens: (71..72, 100..101)
| | | | | | arg #0 (comma): Some(74..75)
| | | | | | arg #0: Variable("LO")@72..74
| | | | | | arg #1 (comma): None
| | | | | | arg #1: BinaryOp@75..100
| | | | | | | op: Eq@84..86
| | | | | | | lhs: BinaryOp@75..84
| | | | | | | | op: Eq@77..79
| | | | | | | | lhs: UnaryOp@75..77
| | | | | | | | | op: Negation@75..76
| | | | | | | | | expr: Variable("b")@76..77
| | | | | | | | rhs: BinaryOp@79..84
| | | | | | | | | op: Sub@82..83
| | | | | | | | | lhs: BinaryOp@79..82
| | | | | | | | | | op: Sub@80..81
| | | | | | | | | | lhs: Variable("O")@79..80
| | | | | | | | | | rhs: Float(4.0)@81..82
| | | | | | | | | rhs: Variable("U")@83..84
| | | | | | | rhs: Paren@86..100
| | | | | | | | parens: (86..87, 99..100)
| | | | | | | | expr: BinaryOp@87..99
| | | | | | | | | op: Eq@95..97
| | | | | | | | | lhs: BinaryOp@87..95
| | | | | | | | | | op: Eq@91..93
| | | | | | | | | | lhs: Bool(true)@87..91
| | | | | | | | | | rhs: Float(0.7)@93..95
| | | | | | | | | rhs: Variable("CE")@97..99
| | rhs: Float(2.0)@103..105
-------
